src/python.rs

```diff
@@ -80,24 +94,40 @@ impl<'j> PythonParser<'j> {
             }
             Peak::Object => {
                 let dict = PyDict::new(py);
+
+                let set_item = |key: PyObject, value: PyObject| {
+                    let r = unsafe { ffi::PyDict_SetItem(dict.as_ptr(), key.as_ptr(), value.as_ptr()) };
+                    // AFAIK this shouldn't happen since the key will always be a string  which is hashable
+                    // we panic here rather than returning a result and using `?` below as it's up to 14% faster
+                    // presumably because there are fewer branches
```

Comment:
thinking on this further it might just be that `panic!` is assumed to be unlikely whereas with branching rustc cannot make any default assumptions, this gap might close with PGO optimization ;)